src/utils/hooks/useListApplications.tsx

```diff
@@ -1,61 +1,108 @@
-import { useEffect, useState } from 'react'
+import { Dispatch, useEffect, useState } from 'react'
+import { ListActions } from '../../contexts/ListState'
 import {
   Application,
   ApplicationStageStatusAll,
   useGetApplicationsQuery,
+  useGetApplicationsStagesQuery,
 } from '../../utils/generated/graphql'
 import { ApplicationDetails } from '../types'
 
 interface UseListApplicationsProps {
   type?: string
+  setListState: Dispatch<ListActions>
 }
 
-const useListApplication = ({ type }: UseListApplicationsProps) => {
-  const [applications, setApplications] = useState<ApplicationDetails[] | undefined>()
+interface ApplicationDetailsMap {
+  [serial: string]: ApplicationDetails
+}
+
+const useListApplication = ({ type, setListState }: UseListApplicationsProps) => {
```

Comment:
Probably want to name it `useListApplications` -- with an `s`, like the filename.

Reply:
Yes. Definitely. I'm changing it now.

Reply:
Done 273e77d